### ISSUE TITLE
create a progress bar at the top right that shows whether or not the course has run all consistency checks (that all collaborators are correct, etc.), and shows progress when they are being run.  Make it possible to click to re-run them.

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#1487]
This -- from a user -- illustrates the confusion that can ensue:
> I encountered a few issues when trying to bulk-add students to my course. Note that I had to do a couple of rounds of adds because new students are continuously adding the course, but I think all of these issues already occurred with the first batch.
> - Some students did not get added to the shared project.
> - Some students could not see/open their master project until I started it for the first time.
> - Some students did not get added to their own course project. Even stranger, a couple of students got removed from their own course project after having successfully used it!
> - When upgrading all students at once, some student projects didn't get upgraded.
> All of these can of course be fixed one case at a time (which I am currently doing), but with >150 students in the course and 15-20 cases that need to be fixed one way or the other, this begins to add up.

---

Perhaps, at some step along the way, one "update" hadn't finished, and he started adding more students, and one ended up interfering with another (?).   With no indicator, it isn't clear to users what to do.

**Comment by kedlaya** [comment]
As the user in question, let me clarify: I'm pretty sure that in my case, this wasn't caused by overlapping operations (the operations were hours apart), but by operations timing out before they were finished (probably because of an interruption to my Internet conneciton).

**Comment by williamstein** [comment]
That makes sense. Moreover, we can have a flag that gets set when the "ensure everything is consistent" process runs to completion **successfully**.  If it fails, it'll be marked as having failed, the UI will show this,  and the user can click to try again.

**Comment by kedlaya** [comment]
Whatever the root cause is (and this might be difficult to pin down), making it trivial for a course admin to both detect and fix the problem would be a big help!

**Comment by kedlaya** [comment]
Additional data: I just added a student to my project and got a timeout while creating projects. As instructed, I closed and reopened the course file, and that seems to have correctly created the student's project and added the student to the shared project. It did not automatically upgrade the student's project, but I suppose that is not something one can do automatically because it depends on a nontrivial condition (namely having the upgrade available).
